### ISSUE TITLE
Not able to make curl-7.54.0

### ISSUE THREAD

**Issue by baali** [curl/curl#1501]
I am trying to compile curl-7.54.0 
I have installed `OpenSSL 1.1.0e` and `nghttp2-1.22.0` when I do `make`:

    Making all in lib
    make[1]: Entering directory '/home/pi/Desktop/curl-7.54.0/lib'
    make  all-am
    make[2]: Entering directory '/home/pi/Desktop/curl-7.54.0/lib'
      CC       vtls/libcurl_la-openssl.lo
    vtls/openssl.c: In function ‘Curl_ossl_seed’:
    vtls/openssl.c:279:5: error: implicit declaration of function ‘RAND_egd’ [-Werror=implicit-function-declaration]
         int ret = RAND_egd(data->set.str[STRING_SSL_EGDSOCKET]?
        ^
    cc1: some warnings being treated as errors
    Makefile:2023: recipe for target 'vtls/libcurl_la-openssl.lo' failed
    make[2]: *** [vtls/libcurl_la-openssl.lo] Error 1

### I have tried:
As mentioned in log that its a warning which is getting treated as error, I have tried to disable this during configure with:  .`/configure --disable-werror`  but the issue persists. 

### operating system
Linux raspberrypi 4.4.50-v7+ #970 SMP Mon Feb 20 19:18:29 GMT 2017 armv7l GNU/Linux

**Comment by bagder** [comment]
The `RAND_egd` function doesn't seem to be generally available in OpenSSL 1.1.0+. This indicates that either the configure detection somehow did wrong, or you have two openssl installations and configure checked for RAND_egd in the older openssl libs of yours while when you compile you get to use the newer set.

Ignoring the compiler warning is harmless, sure, but if the function doesn't exist in openssl it will just postpone the error to link time one instead.

**Comment by baali** [comment]
Thank you for the pointer and its indeed the case. I looked closely to the logs of configure and found this:

    configure: Added /usr/local/lib/lib to LD_LIBRARY_PATH
    checking for OpenSSL headers version... 1.1.0 - 0x1010005fL
    checking for OpenSSL library version... 1.0.1
    checking for OpenSSL headers and library versions matching... no
    configure: WARNING: OpenSSL headers and library versions do not match.

I will fix it up, sorry for the trouble.

**Comment by baali** [comment]
Just mentioning, removing already installed version using 

    $ sudo apt-get remove libssl-dev

worked for me.

**Comment by jkaster** [comment]
Thanks @baali! For me on RHEL, 
```$ sudo yum remove openssl-devel```
worked for me

**Comment by MoNTE48** [comment]
I'm trying to compile curl with NDK and get this error.
netrc.c: In function 'Curl_parsenetrc':
  CC       libcurl_la-rand.lo
netrc.c:81:7: error: implicit declaration of function 'getpwuid_r' [-Werror=implicit-function-declaration]
       if(!getpwuid_r(geteuid(), &pw, pwbuf, sizeof(pwbuf), &pw_res)
       ^

The latest working version is 7.53.1

**Comment by jay** [comment]
> netrc.c:81:7: error: implicit declaration of function 'getpwuid_r'

Please start your own issue and do not piggyback on this one. pwd.h should have that declaration and configure should define HAVE_PWD_H on your system so that it is included in netrc.c. Check both of those and if you haven't figured out the issue start a new one with your config.log and someone with the project will take a look

**Comment by edmwagner** [comment]
I am facing this same issue but will not have the option of removing installed openssl.

I tried setting LDFLAGS,CFLAGS CPPFLAGS,LD_LIBRARY_PATH, and PK_CONFIG_PATH but still no luck.

I make install openssl-1.1.1-pre1 to /dev/shm/installs     // I'm testing this scenario.
Then want to install curl that uses the abov e version to the same location /dev/shm/installs.

But I am seeing the above RAND_egd error in cls/openssl.c:428 

This is with curl-7.58.0

Is there some way to force the openssl includes and libs to be picked up from desired location?

**Comment by jay** [comment]
Please start your own issue. It should be as simple as setting LDFLAGS in Linux. I tried just now in Ubuntu 16 LTS and it worked fine:

openssl-1.1.1-pre1
~~~
./config shared no-zlib --prefix=/opt/ssl --openssldir=/opt/ssl
make
sudo make install
~~~

curl-7.58.0
~~~
LDFLAGS="-Wl,-rpath,/opt/ssl/lib -Wl,-rpath,/opt/lib \
  -Wl,-rpath,/usr/local/lib" \
./configure --with-nghttp2=/usr/local --with-ssl=/opt/ssl --prefix=/opt
make
sudo make install
~~~

If you have a problem open a new issue and include your config log. Note the OpenSSL version detection in the curl 7.58 configure script does not recognize OpenSSL 1.1.1 and will wrongly report a header/library mismatch even if there isn't one. That is fixed in 1b27fcf.
